UMLEditor/UMLEditor/Classes/TimeMachine.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace UMLEditor.Classes;
+
+/// <summary>
+/// Class to represent the action history of the UML program
+/// </summary>
+public static class TimeMachine
+{
+    /// <summary>
+    /// Node class to represent a single state in the time machine history.
+    /// </summary>
+    public class Node
+    {
+        /// <summary>
+        /// Reference to previous node
+        /// </summary>
+        public Node? PrevNode;
+        /// <summary>
+        /// Diagram at this node
+        /// </summary>
+        public Diagram? StateDiagram;
+        /// <summary>
+        /// Reference to next node
+        /// </summary>
+        public Node? NextNode;
+    }
+    /// <summary>
+    /// References to the beginning, end, and current state of the time machine
+    /// </summary>
+    public static Node? Head, Tail, Current;
```

Comment:
The TM's members could be initialized at declaration (this would eliminate the need for the Initialize method). These members and Size should be made private in TimeMachine (Size could be a property). Exposing these publicly allows any class outside of TimeMachine to change them and potentially put the TimeMachine in an erroneous state.